sdks/java/core/src/test/java/org/apache/beam/sdk/io/FileIOTest.java

```diff
@@ -221,36 +227,33 @@ public void testMatchWatchForNewFiles() throws IOException, InterruptedException
     assertEquals(PCollection.IsBounded.UNBOUNDED, matchMetadata.isBounded());
     assertEquals(PCollection.IsBounded.UNBOUNDED, matchAllMetadata.isBounded());
 
+    // Copy the files to the "watch" directory, preserving the lastModifiedTime.
     Thread writer =
         new Thread(
             () -> {
               try {
                 Thread.sleep(1000);
-                Files.write(basePath.resolve("first"), new byte[42]);
+                Files.copy(sourcePath.resolve("first"), watchPath.resolve("first"));
                 Thread.sleep(300);
-                Files.write(basePath.resolve("second"), new byte[37]);
+                Files.copy(sourcePath.resolve("second"), watchPath.resolve("second"));
                 Thread.sleep(300);
-                Files.write(basePath.resolve("third"), new byte[99]);
+                Files.copy(sourcePath.resolve("third"), watchPath.resolve("third"));
               } catch (IOException | InterruptedException e) {
                 throw new RuntimeException(e);
               }
             });
     writer.start();
 
+    // We fetch lastModifiedTime from the files in the "source" directory to avoid a race condition
+    // with the writer thread.
     List<MatchResult.Metadata> expected =
         Arrays.asList(
             metadata(
-                basePath.resolve("first"),
-                42,
-                Files.getLastModifiedTime(basePath.resolve("first")).toMillis()),
+                watchPath.resolve("first"), 42, lastModifiedMillis(sourcePath.resolve("first"))),
```

Comment:
Is the lastModifiedTime changed when the sourcePath files are written? Why do we expect this match when the file are written back out a second time in the watchPath dir?

Or is the last modified time copied from the sourcePath files? So we expect it to be the same?

Would you please add a comment in this test clarifying this for the next reader?

Reply:
Pushed a commit to add additional commentary and to explicitly pass `StandardCopyOption.COPY_ATTRIBUTES` to signal that we are requesting attributes like timestamps to be preserved.